src/django_components/__init__.py

```diff
@@ -21,20 +20,28 @@ def autodiscover() -> None:
         autodiscover_modules("components")
 
         # Autodetect a <component>.py file in a components dir
-        component_filepaths = search(search_glob="**/*.py")
+        component_filepaths = search(search_glob="**/*.py").matched_files
         for path in component_filepaths:
             import_file(path)
 
     for path_lib in app_settings.LIBRARIES:
         importlib.import_module(path_lib)
 
 
-def import_file(path: Union[str, Path]) -> None:
-    MODULE_PATH = path
-    MODULE_NAME = Path(path).stem
-    spec = importlib.util.spec_from_file_location(MODULE_NAME, MODULE_PATH)
-    if spec is None:
-        raise ValueError(f"Cannot import file '{path}' - invalid path")
-    module = importlib.util.module_from_spec(spec)
-    sys.modules[spec.name] = module
-    spec.loader.exec_module(module)  # type: ignore
+def import_file(file_path: Path) -> None:
+    # Get the root dir, see https://stackoverflow.com/a/16413955/9788634
+    project_root = os.path.abspath(os.path.dirname(__name__))
+
+    # Derive python import path from the filesystem path
+    # Example:
+    # If project root is `/path/to/project`
+    # And file_path is `/path/to/project/app/components/mycomp.py`
+    # Then the path relative to project root is `app/components/mycomp.py`
+    # Which we then turn into python import path `app.components.mycomp`
+    rel_path = os.path.relpath(file_path, start=project_root)
+    rel_path_without_suffix = str(Path(rel_path).with_suffix(""))
+    module_name = rel_path_without_suffix.replace(os.sep, ".")
```

Comment:
A couple of tests for this logic would be nice I think. You pretty much already have the testcases there in the file. I see support for non-linux OS:es, that could be in the unit test as well.

Reply:
Good point with testing the `import_file` logic. It wasn't as straightforward:

I omitted test non-linux paths, as I was getting "internal error" when I changed the `os.path.sep` to `\`. 

Also, altho I wrote the test for the file import, it works when ran alone, but some other tests are failing when it's all ran in a single run. So I've [marked it as skipped](https://github.com/EmilStenstrom/django-components/pull/435/files#diff-6faab92c5a5f55ed5c40230f2e1878cfe9603debd5af3d9b90c4ee875d5c2e93R129), and instead created a follow-up ticket (https://github.com/EmilStenstrom/django-components/issues/436) regarding tests isolation.

So the only test that actually runs is a test that raises error if the path is invalid 😅

Reply:
```python
def autodiscover() -> None:
    from django_components.app_settings import app_settings
    if app_settings.AUTODISCOVER:
        autodiscover_modules("components")

        component_filepaths = search(search_glob="**/*.py").matched_files
        for path in component_filepaths:
            path_lib = path_to_path_lib(path)  # <- THIS
            importlib.import_module(path_lib)  # <- THIS

    for path_lib in app_settings.LIBRARIES:
        importlib.import_module(path_lib)
```

I had a even simpler idea. What about breaking those three lines out from import_file, and then add a couple of simple unit tests for `path_to_path_lib`?

I think you can mock os.path.sep like this:

```python
with mock.patch("os.path.sep", new="//"):
    print(os.path.sep)
```

Reply:
Thanks, that makes it clearer! Yup, changing the `import_file` into a simple transformation function made things much simpler. Plus I added logger calls.

Also I noticed that previously I was using `os.sep` instead of `os.path.sep`, not sure if that had any effect. Anyway, let me know what you think 🙌